dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java

```diff
@@ -0,0 +1,354 @@
+ * Default high level client to handle API requests in Elastic
+ */
+public class DotRestHighLevelClientProvider extends RestHighLevelClientProvider {
+
+    private static final String ES_CONFIG_DIR = "config";
+    private static final String ES_PATH_HOME = "es.path.home";
+    private static final String ES_PATH_HOME_DEFAULT_VALUE = "WEB-INF/elasticsearch";
+
+    private static final String BASIC_AUTH_TYPE = "BASIC";
+
+    private static final String JWT_AUTH_TYPE = "JWT";
+
+    private RestHighLevelClient client;
+
+    private static SSLContext sslContextFromPem;
+
+    DotRestHighLevelClientProvider() {
+        try {
+            final RestClientBuilder clientBuilder = getClientBuilder(getCredentialsProvider());
+            client = new RestHighLevelClient(clientBuilder);
+        } catch (IOException | GeneralSecurityException e) {
+            Logger.error(DotRestHighLevelClientProvider.class,
+                    "Error setting credentials for Elastic RestHighLevel Client", e);
+        }
+    }
+
+    private BasicCredentialsProvider getCredentialsProvider() throws IOException, GeneralSecurityException {
+        final String esAuthType = getESAuthType();
+
+        //Loading TLS certificates
+        if (Config.getBooleanProperty("ES_TLS_ENABLED", true)) {
+            loadTLSCertificates();
+        }
+
+        final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+
+        //Loading basic credentials if set
+        if (esAuthType.equals(BASIC_AUTH_TYPE)) {
+            credentialsProvider.setCredentials(AuthScope.ANY,
+                    new UsernamePasswordCredentials(
+                            Config.getStringProperty("ES_AUTH_BASIC_USER", null),
+                            Config.getStringProperty("ES_AUTH_BASIC_PASSWORD", null)));
+            Logger.info(DotRestHighLevelClientProvider.class,
+                    "Initializing Elastic RestHighLevelClient using Basic authentication");
+        }
+
+        return credentialsProvider;
+    }
+
+    private RestClientBuilder getClientBuilder(final BasicCredentialsProvider credentialsProvider) {
+        final String esAuthType = getESAuthType();
+
+        final RestClientBuilder clientBuilder = RestClient
+                .builder(new HttpHost(Config.getStringProperty("ES_HOSTNAME", "127.0.0.1"),
+                        Config.getIntProperty("ES_PORT", 9200), "https"))
+                .setHttpClientConfigCallback((httpClientBuilder) -> {
+                    if (sslContextFromPem != null) {
+                        httpClientBuilder
+                                .setSSLContext(sslContextFromPem);
+                    }
+                    httpClientBuilder
+                            .setDefaultCredentialsProvider(credentialsProvider);
+                    return httpClientBuilder;
+                })
+                .setFailureListener(new RestClient.FailureListener() {
+                    public void onFailure(Node node) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'node' is not assigned and could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=3712629)